### PR TITLE
Feature/server error handling

### LINKED ISSUE #55: Research and implement GRADE A error handling for back end

### PR DESCRIPTION
**Pull request by jfogiato**
### Files Changed:
`server.js`

### What Has Changed: 

- Implemented error handling for PUT
- Implemented error handling for POST

### Notes: (includes questions, ideas for the future, anything you want to call out)

None

### Closes: (any issues that it closes)

Closes #55

### CODE REVIEW

File: server/server.js [hunk 1]
```diff
@@ -26,8 +26,15 @@ app.get('/api/v1/entries', async (req, res) => {
 });
 
 app.post('/api/v1/entries', async (req, res) => {
+  const newEntry = req.body;
+
+  for (let parameter of ['user_id', 'date', 'feeling', 'activity', 'journal_entry']) {
```
Comment (espressoGoddess) [inline server/server.js:31]: This is really cool, I'd love to play around with this method sometime, haven't seen it used much.

File: server/server.js [hunk 2]
```diff
@@ -39,9 +46,15 @@ app.put('/api/v1/entries/:id', async (req, res) => {
   try {
     const { id } = req.params;
     const { journal_entry } = req.body;
-    await knex("entries").where({ entry_id: id }).update({ journal_entry: journal_entry });
-      res.status(204).json(`Entry ${id} was updated successfully`);
+    const dbId = await knex("entries").select("entry_id").where({ entry_id: id });
+
+    if (!dbId[0]?.entry_id) {
```
Comment (espressoGoddess) [inline server/server.js:51]: !!! soooo awesome. great work @jfogiato !

File: server/server.js [hunk 3]
```diff
@@ -39,9 +46,15 @@ app.put('/api/v1/entries/:id', async (req, res) => {
...
+      res.status(400).json({message: `An entry with the id of ${id} was not found.`});
+    } else {
+      await knex("entries").where({ entry_id: id }).update({ journal_entry: journal_entry });
```
Comment (espressoGoddess) [inline server/server.js:54]: i just honestly love how you can use await to kinda "pause" js before going to the next line (At least that is how i am understanding it?), correct me if im wrong!

**Review by espressoGoddess — APPROVED** [review]

File: server/server.js (hunk 3 above)
Comment (jfogiato) [reply server/server.js:54]: SAME!!

File: server/server.js (hunk 1 above)
Comment (KatherineBlaine) [reply server/server.js:31]: Such a clean way to handle for missing data!

File: server/server.js (hunk 2 above)
Comment (KatherineBlaine) [reply server/server.js:51]: 5 stars for this and 5 stars for Knex queries. So readable!

**Review by KatherineBlaine — APPROVED** [review]
This is great. Such thorough error handling, love to see it!

File: server/server.js [hunk 4]
```diff
@@ -39,9 +46,15 @@ app.put('/api/v1/entries/:id', async (req, res) => {
...
+      res.status(202).json({message: `Entry ${id} was updated successfully`});
+    }
   } catch (error) {
     console.error(error.message);
   }
-})
\ No newline at end of file
+});
```
Comment (chrissycooper) [inline server/server.js:60]: Killin it! Thanks for your work on this. Really impressive. Thanks so much!!

**Review by chrissycooper — APPROVED** [review]

**Pull request merged**